### PR TITLE
System identification module

### PR DESCRIPTION
**Pull request by dewagter**
Add a CHIRP deflection on any actuator while adding a little bit of random noise to all other actuators.

 - Based on https://github.com/paparazzi/paparazzi/pull/2374, applied all requested changes.
 - Tested on Quadplane airframe.

### CODE REVIEW

File: sw/airborne/math/pprz_random.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * float random_number = rand_uniform();
+ */
+
+#ifndef RANDOM_H
+#define RANDOM_H
+
+#include <std.h>
+#include <stdlib.h>
+#include <math.h>
+#include <stdbool.h>
```
Comment (gautierhattenberger) [inline sw/airborne/math/pprz_random.h:40]: if really needed, move to C file, I don't think it is useful in the header file

File: sw/airborne/modules/system_identification/pprz_chirp.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ * @file "modules/system_identification/pprz_chirp.c"
+ * @author Joost Meulenbeld
+ * Mathematical implementation of the chirp
+ */
+#include "pprz_chirp.h"
+#include "std.h"
+
+// Values for exponential chirp (See ref [2] in the header file). C2 is based on C1 s.t. the frequency range exactly covers the required range
+#define CHIRP_C1 4.0f
+#define CHIRP_C2 1.0f / (exp(CHIRP_C1) - 1)
```
Comment (gautierhattenberger) [inline sw/airborne/modules/system_identification/pprz_chirp.c:30]: use `expf` for float

File: sw/airborne/modules/system_identification/pprz_chirp.c [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    return chirp->current_value;
+  }
+
+  // If the fade-in is finished, the current time t is the time since the fade-in stopped
+  if (chirp->fade_in) {
+    t -= 2 / chirp->f0_hz;
+  }
+
+  if (chirp->exponential_chirp) { // See the book referenced in the header for the equations
+    float exponential = exp(CHIRP_C1 * t / chirp->length_s);
```
Comment (gautierhattenberger) [inline sw/airborne/modules/system_identification/pprz_chirp.c:100]: same here

File: sw/airborne/modules/system_identification/pprz_chirp.h [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+ * [2] Aircraft and Rotorcraft System Identification, 2nd edition by M. Tischler for a derivation of exponential chirp
+ */
+
+#ifndef PPRZ_CHIRP_H
+#define PPRZ_CHIRP_H
+
+#include <math.h>
+#include <std.h>
+#include <stdbool.h>
+#include <stdlib.h>
```
Comment (gautierhattenberger) [inline sw/airborne/modules/system_identification/pprz_chirp.h:57]: only `"std.h` should be needed here, move the rest to C file (if needed)

Comment (dewagter) [reply sw/airborne/modules/system_identification/pprz_chirp.h]: done

File: sw/airborne/modules/system_identification/pprz_chirp.c (hunk 3 above)
Comment (dewagter) [reply sw/airborne/modules/system_identification/pprz_chirp.c]: done

File: sw/airborne/modules/system_identification/pprz_chirp.c (hunk 2 above)
Comment (dewagter) [reply sw/airborne/modules/system_identification/pprz_chirp.c]: done

File: sw/airborne/math/pprz_random.h (hunk 1 above)
Comment (dewagter) [reply sw/airborne/math/pprz_random.h:40]: First removed everything and added only what was strictly needed to compile.

**Comment by dewagter** [comment]
@gautierhattenberger 
Thanks for the review. I totally agree with all points and applied them.

**Comment by joostmeulenbeld** [comment]
@dewagter nice to see this code making it into paparazzi!

**Pull request merged**